forest/src/cli/sync_cmd.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::stringify_rpc_err;
+use actor::EPOCH_DURATION_SECONDS;
+use chain_sync::get_naive_time_now;
+use chrono::naive::NaiveDateTime;
+use chrono::prelude::*;
+use cid::Cid;
+use jsonrpc_v2::Error as JsonRpcError;
+use jsonrpsee::raw::RawClient;
+use jsonrpsee::transport::http::HttpTransportClient as HTC;
+use rpc_client::{check_bad, head, mark_bad, new_client, status, submit_block};
+use std::thread;
+use std::time::Duration;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub enum SyncCommand {
+    #[structopt(
+        name = "mark-bad",
+        about = "Mark the given block as bad, will prevent syncing to a chain that contains it"
+    )]
+    MarkBad {
+        #[structopt(short, long, help = "Block Cid given as string argument")]
+        block_cid: String,
+    },
+
+    #[structopt(
+        name = "check-bad",
+        about = "Check if the given block was marked bad, and for what reason"
+    )]
+    CheckBad {
+        #[structopt(short, long, help = "Block Cid given as string argument")]
+        block_cid: String,
+    },
+
+    #[structopt(
+        name = "submit",
+        about = "Submit newly created block to network through node"
+    )]
+    Submit {
+        #[structopt(short, long, help = "Gossip block as String argument")]
+        gossip_block: String,
+    },
+
+    #[structopt(name = "status", about = "Check sync status")]
+    Status,
+
+    #[structopt(name = "wait", about = "Wait for sync to be complete")]
+    Wait,
+}
+
+fn get_naive_time_zero() -> NaiveDateTime {
+    NaiveDate::from_ymd(1, 1, 1).and_hms(0, 0, 0)
+}
+
+impl SyncCommand {
+    pub async fn run(self) {
+        let mut client = new_client();
+
+        match self {
+            SyncCommand::Status {} => {
+                let response = status(&mut client).await;
+                if let Ok(r) = response {
+                    println!("sync status:");
+                    for (i, active_sync) in r.active_syncs.iter().enumerate() {
+                        println!("Worker {}:", i);
+                        let mut height_diff = 0;
+                        let height = 0;
+
+                        let mut base: Option<Vec<Cid>> = None;
+                        let mut target: Option<Vec<Cid>> = None;
+
+                        if let Some(b) = &active_sync.base {
+                            base = Some(b.cids().to_vec());
+                            height_diff = b.epoch();
+                        }
+
+                        if let Some(b) = &active_sync.target {
+                            target = Some(b.cids().to_vec());
+                            height_diff = b.epoch() - height_diff;
+                        } else {
+                            height_diff = 0;
+                        }
+
+                        println!("\tBase:\t{:?}\n", base.unwrap_or_default());
```

Comment:
```suggestion
                        println!("\tBase:\n{:?}\n", base.unwrap_or_default());
```
I think you wanted a newline there, and are you sure you want the double newlines after each thing? (println adds one by default and you add `\n` at the end